app/src/org/commcare/activities/FormEntryActivity.java

```diff
@@ -1124,22 +1123,21 @@ private void showView(QuestionsView next, AnimationType from, boolean animateLas
         mViewPane.addView(questionsView, lp);
 
         questionsView.startAnimation(mInAnimation);
-
-        FrameLayout header = (FrameLayout)findViewById(R.id.form_entry_header);
-
-        TextView groupLabel = ((TextView)header.findViewById(R.id.form_entry_group_label));
-
-        this.mGroupNativeVisibility = false;
-        FormLayoutHelpers.updateGroupViewVisibility(this, mGroupNativeVisibility, mGroupForcedInvisible);
-
         questionsView.setFocus(this);
 
+        setupGroupLabel();
+    }
+
+    private void setupGroupLabel() {
+        hasGroupLabel = false;
+        FormLayoutHelpers.updateGroupViewVisibility(this, false, shouldHideGroupLabel);
         SpannableStringBuilder groupLabelText = questionsView.getGroupLabel();
 
         if (groupLabelText != null && !groupLabelText.toString().trim().equals("")) {
+            TextView groupLabel = (TextView)findViewById(R.id.form_entry_group_label);
             groupLabel.setText(groupLabelText);
-            this.mGroupNativeVisibility = true;
-            FormLayoutHelpers.updateGroupViewVisibility(this, mGroupNativeVisibility, mGroupForcedInvisible);
+            hasGroupLabel = true;
+            FormLayoutHelpers.updateGroupViewVisibility(this, true, shouldHideGroupLabel);
```

Comment:
the line above sets it to `true` so it is always true. Mostly did this because the linter was complaining.